library/core/src/ptr/mod.rs

```diff
@@ -777,8 +777,17 @@ where
 
 /// Convert a reference to a raw pointer.
 ///
-/// This is equivalent to `r as *const T`, but is a bit safer since it will never silently change
-/// type or mutability, in particular if the code is refactored.
+/// For `r: &T`, `from_ref(r)` is equivalent to `r as *const T`, but is a bit safer since it will
+/// never silently change type or mutability, in particular if the code is refactored.
+///
+/// The caller must ensure that the pointee outlives the pointer this function returns, or else it
+/// will end up pointing to garbage.
```

Comment:
I suggest replacing "pointing to garbage" with "dangling". I also suggest adding here a concrete example from the clippy experience:

For example, `as_ref(r)` is not equivalent to `r as *const T` when lifetime extension is in effect:
```
let p = &foo() as *const T; // p is valid because of lifetime extension
```
to
```
let p = ptr::from_ref(&foo()); // Dangerous: p is dangling.
```
.